.circleci/config.yml

```diff
@@ -12,15 +12,29 @@ jobs:
       - checkout
       - run:
           name: Validate Terraform configurations
-          command: find . -type f -name "*.tf" -exec dirname {} \;|sort -u | while read m; do (terraform validate -check-variables=false "$m" && echo "√ $m") || exit 1 ; done
+          command: |
+            changed_resources=$(git diff --no-commit-id --name-only -r origin/master...HEAD | grep /resources/ | sort | uniq) || true
+            for _f in ${changed_resources}; do
+              dirname ${_f} | while read m; 
+                do 
+                  (terraform validate -check-variables=false "$m" && echo "√ $m") || exit 1 ;
+                done 
+            done 
       - run:
           name: Check if Terraform configurations are properly formatted
-          command: tf=$(terraform fmt -write=false -diff); if [[ -n "$tf" ]]; then printf "$tf\n\n\nSome terraform files failed syntax validation, run 'terraform fmt' to fix\n\n"; exit 1; fi
+          command: |
+            changed_resources=$(git diff --no-commit-id --name-only -r origin/master...HEAD | grep /resources/ | sort | uniq) || true
+            for _f in ${changed_resources}; do
+              tf=$(terraform fmt -write=false -diff ${_f}); 
+              if [[ -n "$tf" ]]; then 
```

Comment:
To reduce complexity, we could treat this like the other check above. From `terraform fmt -help`:
```
  -check=false     Check if the input is formatted. Exit status will be 0 if all input is properly formatted and non-zero otherwise.
```
We can use that (perhaps with `-diff` as well, in order to show the actual problem) and end up with something like:
```
(terraform fmt -check -diff ${_f}) || exit 1;
```
That would make this snippet very similar to the validate snippet above.

Reply:
agree with this as well to keep it simple and similar as above, updated